protostar/configuration_file/configuration_file.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import List, Optional, Union
+
+from typing_extensions import Protocol
+
+from protostar.protostar_exception import ProtostarException
+from protostar.utils.protostar_directory import VersionType
+
+PrimitiveTypesSupportedByConfigurationFile = Union[str, int, bool]
+
+
+class ConfigurationFile(Protocol):
+    def get_min_protostar_version(self) -> Optional[VersionType]:
```

Comment:
`get` -> `fetch` wdyt? so that it won'd look like these calls are cheap

Reply:
Configuration is going to be cached, so it will be cheap.

Reply:
then why not using `@property` then? :)

Reply:
`ConfigurationFile` is meant to be used as an object, not as a data structure. ConfigurationFile will handle converting relative to absolute paths and reduce the tension between configuration file structure and application needs.